tutorials/vue/vue-d3.md

```diff
@@ -0,0 +1,148 @@
+which is built upon `D3.js`. `vue-d3-charts` does not support vue2 and so
+you will need to import `D3.js`.
+
+As of this writing, development for vue-d3-charts is currently frozen as the head maintainer does not have the time
+to continue improving this library. However, individuals are free to contribute [here](https://github.com/Saigesp/vue-d3-charts).
+We recommend using other data visualizing libraries as any issues you may run into while using
+this library will most likely not be solved for a good period of time.
+
+Here is an example of using `D3.js` in Vue2 to create a bar graph.
+A great thing about `D3.js` is that you can import and parse a csv file from an online source to use directly in your visualizations.
+Below is the code:
+
+```html
+<script src="https://d3js.org/d3.v6.js"></script>
+<script src="https://unpkg.com/vue"></script>
+<div class = "app">{{message}}
+<bargraph></bargraph>
+</div>
+<script type = "text/javascript">
+Vue.component('bargraph', {
+    mounted(){
+        this.retrieveGraph();
+    },
+    methods: {
+        retrieveGraph: function(){
+            let margin = ({top: 30, right: 30, bottom: 70, left: 60});
+            let width = 460 - margin.left - margin.right;
+            let height = 400 - margin.top - margin.bottom;
+            var svg = d3.select("#example")
+                .append("svg")
+                .attr("width", width + margin.left + margin.right)
+                .attr("height", height + margin.top + margin.bottom)
+                .append("g")
+                .attr("transform",
+                "translate(" + margin.left + "," + margin.top + ")");
+            this.getData().then((info) => {
+            // Add X axis
+            var x = d3.scaleBand()
+            .range([ 0, width ])
+            .domain(info.map(function(d) { return d.Country; }))
+            .padding(0.2);
+            svg.append("g")
+            .attr("transform", "translate(0," + height + ")")
+            .call(d3.axisBottom(x))
+            .selectAll("text")
+                .attr("transform", "translate(-10,0)rotate(-45)")
+                .style("text-anchor", "end");
+            // Add Y axis
+            var y = d3.scaleLinear()
+            .domain([0, 13000])
+            .range([ height, 0]);
+            svg.append("g")
+            .call(d3.axisLeft(y));
+            // Bars
+            svg.selectAll("mybar")
+            .data(info)
+            .enter()
+            .append("rect")
+                .attr("x", function(d) { return x(d.Country); })
+                .attr("y", function(d) { return y(d.Value); })
+                .attr("width", x.bandwidth())
+                .attr("height", function(d) { return height - y(d.Value); })
+                .attr("fill", "#69b3a2")
+            });
+                    },
+        getData: async function(){
+            return await d3.csv("https://raw.githubusercontent.com/holtzy/data_to_viz/master/Example_dataset/7_OneCatOneNum_header.csv");
```

Comment:
I'd just inline the data into the test case as a [POJO](https://masteringjs.io/tutorials/fundamentals/pojo) rather than fetching it as a CSV. Easier to read and less code to write